src/HotChocolate/Core/src/Types/Types/Scalars/TimeSpanType.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Generic;
+using System.Xml;
+using HotChocolate.Language;
+using HotChocolate.Properties;
+
+#nullable enable
+
+namespace HotChocolate.Types
+{
+    /// <summary>
+    /// The TimeSpan scalar type represented in two formats:
+    /// <see cref="TimeSpanFormat.ISO_8601"/> and <see cref="TimeSpanFormat.DOT_NET"/>
+    /// </summary>
+    public sealed class TimeSpanType
+        : ScalarType<TimeSpan, StringValueNode>
+    {
+        private static readonly Dictionary<TimeSpanFormat, Func<TimeSpan, string>> _timeSpanToString =
+            new Dictionary<TimeSpanFormat, Func<TimeSpan, string>>
+            {
+                {TimeSpanFormat.ISO_8601, timeSpan => XmlConvert.ToString(timeSpan)},
+                {TimeSpanFormat.DOT_NET, timeSpan => timeSpan.ToString("c")}
+            };
+
```

Comment:
Hm. I wonder if it makes sense to expose this abstraction to the user. 
So instead of having a Enum just having an interfaces with some default implementations:

```csharp
interface ITimeSpanFormat {
    string Format(TimeSpan timeSpan);
    bool   TryParse(string serialized, out TimeSpan? value);
}

public static class TimeSpanFormatDefaults {
    public static ITimeSpanFormat Iso = new TimeSpanFormatDefaults.IsoFormatter();
    public static ITimeSpanFormat DotNet = new TimeSpanFormatDefaults.DotNetFormatter();
    .... 
}
```

This way we could also omit `TryDeserialize_ISO_8601` and `TryDeserialize_DOT_NET`

What do you think? Overengieenring?

Reply:
The user should have the possibility to change the format. I have to take a look where this should happens.

Reply:
I think we should implement this fixed. Also, we should not put it in a dictionary .... this is not fast. Scalars have a fixed behavior and should be fast. I someone wants to implement a new formatted he/she can implement a new scalar.